### PR TITLE
HOCS-3780 - Add diff based test for COMP bpmn files

### PR DESCRIPTION
**Pull request by fewtrees-ho**
Add a test to ensure we have checked and confirmed the differences between COMP & COMP2 BPMN files

### CODE REVIEW

File: src/test/scripts/diff_COMP_COMP2.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  echo "             is then used in check_COMP_COMP2.sh as part of the Gradle test process."
+  echo "Usage:"
+  echo "COMP_COMP2_diff <source_file> <target_dir>"
+  echo "  <source_file> - The COMP bpmn file"
+  echo "  <target_dir>  = The target directory to store the diff file"
+}
+# $1 - COMP source file inc path
+# $2 - target dir path for output
+
+if [ -z "$1" ]; then
```
Comment (LiamMacP) [inline src/test/scripts/diff_COMP_COMP2.sh:14]: Could be collapsed into one?

File: src/test/scripts/check_COMP_COMP2.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  # Build the diff output filename
+  diffImage=`echo "$(basename $f)-$(basename $comp2file).diff" | sed 's/\.bpmn//g'`
+  rm -f diff.tmp
+  echo "Checking $f $comp2file...\c"
+  diff $f $comp2file > diff.tmp
+  diff diff.tmp $2/$diffImage
+  exitCode=$?
+  rm -f diff.tmp
+  if [ $exitCode -ne 0 ]; then
+     echo "Unexpected error found"
```
Comment (gardnersam) [inline src/test/scripts/check_COMP_COMP2.sh:19]: Could this print out what file it failed on?

**Comment by fewtrees-ho** [comment]
It does as it's going through..

Each file is printed out and them appended with an 'OK' or details of the error 🙂
________________________________

File: src/test/scripts/check_COMP_COMP2.sh (hunk 2 above)
Comment (gardnersam) [reply src/test/scripts/check_COMP_COMP2.sh:19]: Yep so it does, I should have looked 5 lines up!

**Review by gardnersam — APPROVED** [review]
Looks good, a good solution for enforcing keeping the stages in line

**Pull request merged**